code/game/objects/structures/window.dm

```diff
@@ -336,13 +338,14 @@
 	update_nearby_icons()
 
 /obj/structure/window/proc/spawnDebris(location)
+	var/datum/material/glass_material_ref = GET_MATERIAL_REF(glass_material_datum)
 	. = list()
-	. += new /obj/item/shard(location)
-	. += new /obj/effect/decal/cleanable/glass(location)
+	. += new glass_material_ref.shard_type(location)
+	. += new glass_material_ref.debris_type(location)
```

Comment:
Should there be support for a material datum not producing shards?

Reply:
I can look into adding it if you think it'd be useful, but at present all glass types have an associated shard.

Reply:
i'm concerned about null-safety in case someone decides to add a new debris_type var that isn't supported and we runtime out of spawning debris rather than account for it. if we could support it, that'd be nice, but i just have an odd feeling that if we don't do something about it now we're sitting on somethiing that will york on us eventually